### ISSUE TITLE
[BUG] WPFilesystem::havePlugin erases plugins folder if $path is a single file plugin

### ISSUE THREAD

**Issue by Luc45** [lucatume/wp-browser#459]
This works as expected:
`\Codeception\Module\WPFilesystem::havePlugin('foo/bar.php', $code);`

This will delete your plugins folder:
`\Codeception\Module\WPFilesystem::havePlugin('foo.php', $code);`

The reason for that is that `havePlugin` deletes the parent folder of the plugin after the test ends, using `dirname()`.
If the plugin is a single-file plugin with no folder of it's own, the parent folder is the Plugins folder.

Unfortunately I don't have enough bandwith right now to open a decent PR for this - I have to finish something urgent in the next 50 minutes.

Thanks.

**Comment by lucatume** [comment]
@Luc45  thanks for reporting the issue: it's clearly a bug and removing the parent single-file plugin directory is *not* the correct approach.

I will look into this and provide a fix as soon as I have the bandwidth.

**Comment by Luc45** [comment]
Hi Luca,

I'll give it a shot at this PR. I had some annoying health issues this past week, fever, cough and body pain - luckily it's not COVID-19.

Let's see if I can still get wp-browser up and running for a ride.

**Comment by lucatume** [comment]
This escalated quickly! 
No worries about this, you should be able to build the dev environment used in CI with:

```bash
_build/vendor_prepare.sh 5.6 4.0
make ready
make dev
```

**Comment by Luc45** [comment]
Hi @lucatume,

I'm starting to have some free time as of today. Setting up the environment went surprisingly well, I've just faced some minor issues with `vendor/wordpress/wordpress` being owned by root and being empty, but I've got around that setting the permissions manually and running a `wp core download` in that folder, mimicking `make refresh_wp`.

I'm thinking about tackling this this way:

- Restricting the accepted string as a plugin filename to be either: 'foo.php' or 'foo/foo.php'

If 'foo.php', delete the file. If 'foo/foo.php', delete the `basedir`.

`foo/bar/baz.php` will throw an exception.

Sounds good?

**Comment by lucatume** [comment]
@Luc45 I guessed you would have some issues with the file modes as the script I've set up for Mac where that is not an issue, I apologize for the annoyance.

I would not limit the use of the `havePlugin`, and `haveMuPlugin`, methods to accept only one-level-deep plugin structures; I would parse the input path, e.g. `foo.php` or `foo/plugin.php` or `foo/bar/plugin.php` to assess if the root component is a file, as is the case for `foo.php` or not (as is the case for the other 2 examples) and decide there and then what needs to be deleted.
